typescript/sdk/src/core/EvmIcaModule.ts

```diff
@@ -0,0 +1,73 @@
+import { ProtocolType, rootLogger } from '@hyperlane-xyz/utils';
+
+import { HyperlaneContracts } from '../contracts/types.js';
+import { InterchainAccountDeployer } from '../middleware/account/InterchainAccountDeployer.js';
+import { InterchainAccountFactories } from '../middleware/account/contracts.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { EthersV5Transaction } from '../providers/ProviderType.js';
+import { ProxiedRouterConfig } from '../router/types.js';
+import { ChainNameOrId } from '../types.js';
+
+import { HyperlaneModule, HyperlaneModuleArgs } from './AbstractHyperlaneModule.js';
+
+export type InterchainAccountConfig = ProxiedRouterConfig;
+
+export class EvmIcaModule extends HyperlaneModule<
+  ProtocolType.Ethereum,
+  InterchainAccountConfig,
+  HyperlaneContracts<InterchainAccountFactories>
```

Comment:
imo this should be `HyperlaneAddresses` instead of contracts?

Reply:
or the constructor should take addresses instead of contracts

Reply:
yup that's correct - need to update the ISM module to do this too

Reply:
Done!